src/Compilers/VisualBasic/Test/Emit/CodeGen/CodeGenTuples.vb

```diff
@@ -17423,6 +17423,222 @@ implicit operator AA
+            End Get
+        End Property
+
+        Public ReadOnly Property Current1 As Object Implements IEnumerator.Current
+            Get
+                Return Current
+            End Get
+        End Property
+
+        Public Sub Dispose() Implements IDisposable.Dispose
+        End Sub
+
+        Public Function MoveNext() As Boolean
+            index += 1
+            Return index &lt; _backing.Length
+        End Function
+
+        Public Sub Reset()
+            Throw New NotSupportedException()
+        End Sub
+
+        Private Function IEnumerator_MoveNext() As Boolean Implements IEnumerator.MoveNext
+            Return MoveNext()
+        End Function
+
+        Private Sub IEnumerator_Reset() Implements IEnumerator.Reset
+            Throw New NotImplementedException()
+        End Sub
+    End Class
+
+    Public Function GetEnumerator() As IEnumerator(Of (key As Integer, val As Integer)) Implements IEnumerable(Of (key As Integer, val As Integer)).GetEnumerator
+        Return New Inner(_backing)
+    End Function
+
+    Private Function IEnumerable_GetEnumerator() As IEnumerator Implements IEnumerable.GetEnumerator
+        Return New Inner(_backing)
+    End Function
+    End Structure
+
+
+    </file>
+</compilation>,
+additionalRefs:=s_valueTupleRefs)
+
+            CompileAndVerify(comp, symbolValidator:=Sub(m)
+                                                        Dim c = m.GlobalNamespace.GetTypeMember("C")
+                                                        Assert.Equal(1, c.TypeParameters.Length)
+                                                        Dim param = c.TypeParameters(0)
+                                                        Assert.Equal(1, param.ConstraintTypes.Length)
+                                                        Dim constraint = Assert.IsAssignableFrom(Of NamedTypeSymbol)(param.ConstraintTypes(0))
+                                                        Assert.True(constraint.IsGenericType)
+                                                        Assert.Equal(1, constraint.TypeArguments.Length)
+                                                        Dim typeArg As TypeSymbol = constraint.TypeArguments(0)
+                                                        Assert.True(typeArg.IsTupleType)
+                                                        Assert.Equal(2, typeArg.TupleElementTypes.Length)
+                                                        Assert.All(typeArg.TupleElementTypes,
+                                                                        Sub(t) Assert.Equal(SpecialType.System_Int32, t.SpecialType))
+                                                        Assert.False(typeArg.TupleElementNames.IsDefault)
+                                                        Assert.Equal(2, typeArg.TupleElementNames.Length)
+                                                        Assert.Equal({"key", "val"}, typeArg.TupleElementNames)
```

Comment:
Consider removing the two checks (is default and length) above, as they are redundant.